Src/FluentAssertions/Equivalency/EnumerableEquivalencyValidator.cs

```diff
@@ -137,9 +179,20 @@ private string[] TryToMatch<T>(object subject, T expectation, int expectationInd
             }
         }
 
-        private void StrictlyMatchAgainst<T>(object[] subjects, T expectation, int expectationIndex)
+        private bool StrictlyMatchAgainst<T>(object[] subjects, T expectation, int expectationIndex)
         {
-            parent.AssertEqualityUsing(context.CreateForCollectionItem(expectationIndex.ToString(), subjects[expectationIndex], expectation));
+            using (var scope = new AssertionScope())
+            {
+                object subject = subjects[expectationIndex];
+                string indexString = expectationIndex.ToString();
+                var equivalencyValidationContext = context
```

Comment:
🙃 In this code-base, we prefer explicit types, unless the type itself is clear already.

Reply:
Ok, fixed.